src/VisualStudio/Core/Def/RoslynPackage.cs

```diff
@@ -90,6 +92,28 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
             TrackBulkFileOperations();
         }
 
+        private async Task InitializeWorkspaceFailureOutputWindowAsync()
+        {
+            var threadingContext = _componentModel.GetService<IThreadingContext>();
+            var asyncListenerProvider = _componentModel.GetService<IAsynchronousOperationListenerProvider>();
+            var asyncListener = asyncListenerProvider.GetListener(FeatureAttribute.WorkspaceFailureLogger);
+
+            var brokeredServiceContainer = await this.GetServiceAsync<SVsBrokeredServiceContainer, IBrokeredServiceContainer>().ConfigureAwait(false);
+            Assumes.Present(brokeredServiceContainer);
+            var serviceBroker = brokeredServiceContainer.GetFullAccessServiceBroker();
+            var serviceBrokerClient = new ServiceBrokerClient(serviceBroker, threadingContext.JoinableTaskFactory);
+
+            _workspace.WorkspaceFailed += (sender, eventArgs) =>
+            {
+                var asyncToken = asyncListener.BeginAsyncOperation(nameof(InitializeWorkspaceFailureOutputWindowAsync));
+                Task.Run(async () =>
+                {
+                    using var outputChannelStore = await serviceBrokerClient.GetProxyAsync<IOutputChannelStore>(VisualStudioServices.VS2019_4.OutputChannelStore).ConfigureAwait(false);
```

Comment:
What is this `using` syntax? I've never seeing a `using` block that was missing a parentheses around the expression to be disposed.
Regardless, this looks buggy. It appears you're acquiring a rental and disposing it *immediately*, only to then try to use the proxy that you just released the rental for. You should make sure the using block includes all uses of the rented proxy before disposing the rental.

Reply:
It's a new C# 8 feature, so it's possible I'm using it incorrectly - https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/using

But as far as I understand it, the lifetime of the variable defined in the implicit using will be the scope where it is declared, and once it goes out will be disposed of properly.

Reply:
(Actual non-proposal link) - https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-8#using-declarations

Reply:
That's *awesome*. Thanks for teaching me more about C# 8. In that case what you're doing here looks fine. I'm going to go simplify all my code now. :)